docs/DafnyRef/Attributes.md

```diff
@@ -344,136 +302,206 @@ recursion was explicitly requested.
-       in which implementations are verified (default: N = 1).
-
-     {:id <string>}
-       Assign a unique ID to an implementation to be used for verification
-       result caching (default: "<impl. name>:0").
-
-     {:timeLimit N}
-       Set the time limit for a given implementation.
-
-  ---- On functions ----------------------------------------------------------
-
-     {:builtin "spec"}
-     {:bvbuiltin "spec"}
-       Rewrite the function to built-in prover function symbol 'fn'.
-
-     {:inline}
-     {:inline true}
-       Expand function according to its definition before going to the prover.
-
-     {:never_pattern true}
-       Terms starting with this function symbol will never be
-       automatically selected as patterns. It does not prevent them
-       from being used inside the triggers, and does not affect explicit
-       trigger annotations. Internally it works by adding {:nopats ...}
-       annotations to quantifiers.
-
-     {:identity}
-     {:identity true}
-       If the function has 1 argument and the use of it has type X->X for
-       some X, then the abstract interpreter will treat the function as an
-       identity function.  Note, the abstract interpreter trusts the
-       attribute--it does not try to verify that the function really is an
-       identity function.
-
-  ---- On variables ----------------------------------------------------------
-
-     {:existential true}
-       Marks a global Boolean variable as existentially quantified. If
-       used in combination with option /contractInfer Boogie will check
-       whether there exists a Boolean assignment to the existentials
-       that makes all verification conditions valid.  Without option
-       /contractInfer the attribute is ignored.
-
-  ---- On assert statements --------------------------------------------------
-
-     {:subsumption n}
-       Overrides the /subsumption command-line setting for this assertion.
-
-     {:split_here}
-       Verifies code leading to this point and code leading from this point
-       to the next split_here as separate pieces.  May help with timeouts.
-       May also occasionally double-report errors.
-
-  ---- The end ---------------------------------------------------------------
 
+The quantifier is translated into the following Boogie:
+
+```
+(forall q$heap#8: Heap, u#5: int ::
+    {:heapQuantifier}
+    $IsGoodHeap(q$heap#8) && ($Heap == q$heap#8 || $HeapSucc($Heap, q$heap#8))
+       ==> $Unbox(Apply1(TInt, TInt, f#0, q$heap#8, $Box(u#5))): int == u#5 + r#0);
 ```
 
-However a scan of Boogie's sources shows it checks for the
+What this is saying is that the quantified expression, `f(u) == u + r`,
+which may depend on the heap, is also valid for any good heap that is either the
+same as the current heap, or that is derived from it by heap update operations.
+
+### 22.5.2. `{:induction}` {#sec-induction-quantifier}
+See [`{:induction}`](#sec-induction).
+
+### 22.5.3. `{:layerQuantifier}`
```

Comment:
The word "layer" is old. We refer to this as "fuel" now. I suggest changing the text to use the word "fuel" instead of "layer", perhaps with a comment like `the word "layer" really refers to "fuel"`.

Reply:
I reopened the issue
https://github.com/dafny-lang/dafny/issues/35